core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -662,3 +670,47 @@ private Logger logger() {
         return DockerLoggerFactory.getLogger(COMPOSE_EXECUTABLE);
     }
 }
+
+@Slf4j
+@UtilityClass
+class DockerComposePrecondition {
+    private static final String CONTAINER_NAME_OPTION = "container_name";
+    private static final String VERSION_KEY = "version";
+    private static final Yaml YAML = new Yaml(new Constructor(Map.class));
+
+    static void checkNotContainUnsupportedComposeOptions(List<File> composeFiles) {
+        composeFiles.forEach(file -> checkArgument(!containsUnsupportedComposeOptions(file),
+            String.format("Compose file %s contains '%s' option which is not supported by container.",
+                file,
+                CONTAINER_NAME_OPTION)));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static boolean containsUnsupportedComposeOptions(File file) {
+        try {
+            return Optional.of(YAML.load(FileUtils.openInputStream(file)))
+                .map(yaml -> getServiceOptions((Map<String, Object>) yaml)
+                    .anyMatch(properties -> ((Map<String, Object>) properties).containsKey(CONTAINER_NAME_OPTION)))
+                .orElse(false);
+        } catch (ConstructorException e) {
+            log.warn(String.format("Unable to read yaml structure from compose file %s.", file.getName()), e);
+            return false;
+        } catch (IOException e) {
+            throw new ContainerLaunchException(String.format("Unable to read compose file %s.", file.getName()), e);
+        } catch (Exception e) {
+            log.warn(e.getMessage(), e);
+            return false;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static Stream<Object> getServiceOptions(Map<String, Object> yaml) {
+        if (yaml.containsKey(VERSION_KEY)) {
+            return ((Map) yaml.get("services")).values()
```

Comment:
what if Compose version 5 will have `services` as a list? Please add a check and handle gracefully

Reply:
it so, it will fall in catch block which returns false
is it really good idea to handle that king of possible change in docker compose format?

Reply:
that's our code. We should make sure that our code is protected from future changes.
Since the source of changes is not under our control, we should verify every assumption we make and react if something goes not as we expected

Reply:
just to clarify - we have an if statement where well known (not v1) structure of docker compose file is processed. Return at line 713 is for v1. If Compose version 5 has services as a list it will fail at line 709 and will be processed by catch block. 
What's the reason to have another branch for fictional docker compose structure? It's already handled by try catch at line 690. 

Probably i miss a case where mentioned try catch does not work.

Reply:
I would not rely on that exception when we can do it gracefully